curvefs/src/common/s3util.h

```diff
@@ -32,11 +32,16 @@ namespace s3util {
 inline std::string GenObjName(uint64_t chunkid, uint64_t index,
                               uint64_t compaction, uint64_t fsid,
                               uint64_t inodeid) {
-    return std::to_string(fsid) + "_" + std::to_string(inodeid) + "_" +
+    return std::to_string(fsid) + "/" +
```

Comment:
Two suggestions:
1. Maybe we don't need the directory prefix as the object name？
2. maybe we can use hash or % replace ？

Reply:
Hi, Adding directory prefix has three advantages.
1:  improve the s3 data storage sync performance when we sync different cloud provider using the tools rclone
2: improve s3 write performance when deploying on aws as the [doc](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html) suggested
3: improve disk cache write performance.
So I suggerest using the directory prefix.

Reply:
> Hi, Adding directory prefix has three advantages. 1: improve the s3 data storage sync performance when we sync different cloud provider using the tools rclone 2: improve s3 write performance when deploying on aws as the [doc](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html) suggested 3: improve disk cache write performance as writing 1000w file into 1 directory maybe has poor write performance. So I suggerest using the directory prefix.


yeah, you are right. lets do it.  but can we use hash or % replace `inodeid/1000` ? then the files will more evenly distributed,？

Reply:
> > Hi, Adding directory prefix has three advantages. 1: improve the s3 data storage sync performance when we sync different cloud provider using the tools rclone 2: improve s3 write performance when deploying on aws as the [doc](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html) suggested 3: improve disk cache write performance as writing 1000w file into 1 directory maybe has poor write performance. So I suggerest using the directory prefix.
> 
> yeah, you are right. lets do it. but can we use hash or % replace `inodeid/1000` ? then the files will more evenly distributed,？

yep, maybe we can make it an optional as [juicefs](https://github.com/juicedata/juicefs/blob/main/pkg/chunk/cached_store.go#L74) does.

Reply:
good job, lets do it.

Reply:
> > > Hi, Adding directory prefix has three advantages. 1: improve the s3 data storage sync performance when we sync different cloud provider using the tools rclone 2: improve s3 write performance when deploying on aws as the [doc](https://docs.aws.amazon.com/AmazonS3/latest/userguide/optimizing-performance.html) suggested 3: improve disk cache write performance as writing 1000w file into 1 directory maybe has poor write performance. So I suggerest using the directory prefix.
> > 
> > 
> > yeah, you are right. lets do it. but can we use hash or % replace `inodeid/1000` ? then the files will more evenly distributed,？
> 
> yep, maybe we can make it an optional as [juicefs](https://github.com/juicedata/juicefs/blob/main/pkg/chunk/cached_store.go#L74) does.

good job, lets do it